### PR TITLE
migrate from failure to anyhow

### PR DESCRIPTION
**Pull request by syphar**
I would love to get rid of some failure/anyhow compatibility code in docs.rs, and failure is unmaintained anyways.

### CODE REVIEW

**Review by jyn514 — APPROVED** [review]

**Comment by syphar** [comment]
~~btw, the test failures are odd, if feels like these are not related to my change, but I don't know rustwide well enough to be sure.~~ 

~~If the tests need fixing I'll dig into it.~~

**Comment by syphar** [comment]
@pietroalbini only pinging you because I heard you're still the main maintainer? 
I'm happy to rework this PR to use more specific error types (for example with `thiserror`), if that is what you prefer. 

My motivation is mostly to get rid of the ugly failure-to-anyhow conversion code in docs.rs.

File: tests/utils/mod.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-use failure::Error;
+use anyhow::Result;
```
Comment (GuillaumeGomez) [inline tests/utils/mod.rs]: Let's maybe avoid importing a type under the name of an existing one? Please either rename it or not import it and always refer to it with `anyhow::Result`.

File: tests/issue_30.rs [hunk 2]
```diff
@@ -1,11 +1,11 @@
-use failure::Error;
+use anyhow::Result;
```
Comment (GuillaumeGomez) [inline tests/issue_30.rs]: Same everywhere of course. :)

File: src/crates/git.rs [hunk 3]
```diff
@@ -2,7 +2,7 @@ use super::CrateTrait;
 use crate::cmd::{Command, ProcessLinesActions};
 use crate::prepare::PrepareError;
 use crate::Workspace;
-use failure::{Error, ResultExt};
+use anyhow::{Context as _, Result};
```
Comment (GuillaumeGomez) [inline src/crates/git.rs]: Still there. ;)

File: src/crates/local.rs [hunk 4]
```diff
@@ -75,10 +74,10 @@ fn copy_dir(src: &Path, dest: &Path) -> Result<(), Error> {
 
 #[cfg(test)]
 mod tests {
-    use failure::Error;
+    use anyhow::Result;
```
Comment (GuillaumeGomez) [inline src/crates/local.rs]: You can remove this import I think now.

File: src/crates/registry.rs [hunk 5]
```diff
@@ -1,6 +1,6 @@
 use super::CrateTrait;
 use crate::Workspace;
-use failure::{Error, ResultExt};
+use anyhow::{Context as _, Result};
```
Comment (GuillaumeGomez) [inline src/crates/registry.rs]: Still there too.

File: src/native/windows.rs [hunk 6]
```diff
@@ -1,6 +1,6 @@
 use super::CurrentUser;
 use crate::cmd::KillFailedError;
-use failure::Error;
+use anyhow::{anyhow, Result};
```
Comment (GuillaumeGomez) [inline src/native/windows.rs]: Here too.

File: src/prepare.rs [hunk 7]
```diff
@@ -1,6 +1,6 @@
 use crate::cmd::Command;
 use crate::{build::CratePatch, Crate, Toolchain, Workspace};
-use failure::{Error, Fail, ResultExt};
+use anyhow::{Context as _, Result};
```
Comment (GuillaumeGomez) [inline src/prepare.rs]: Here as well.

File: src/tools/rustup.rs [hunk 8]
```diff
@@ -2,7 +2,7 @@ use crate::cmd::{Binary, Command, Runnable};
 use crate::toolchain::MAIN_TOOLCHAIN_NAME;
 use crate::tools::{Tool, RUSTUP};
 use crate::workspace::Workspace;
-use failure::{Error, ResultExt};
+use anyhow::{Context as _, Result};
```
Comment (GuillaumeGomez) [inline src/tools/rustup.rs]: Here too.

File: src/workspace.rs [hunk 9]
```diff
@@ -2,7 +2,7 @@ use crate::build::BuildDirectory;
 use crate::cmd::{Command, SandboxImage};
 use crate::inside_docker::CurrentContainer;
 use crate::Toolchain;
-use failure::{Error, ResultExt};
+use anyhow::{Context as _, Result};
```
Comment (GuillaumeGomez) [inline src/workspace.rs]: Here++.

File: tests/buildtest/inside_docker.rs [hunk 10]
```diff
@@ -1,6 +1,6 @@
 #![cfg_attr(windows, allow(unused))]
 
-use failure::{Error, ResultExt};
+use anyhow::{Context, Result};
```
Comment (GuillaumeGomez) [inline tests/buildtest/inside_docker.rs]: Here².

File: tests/buildtest/mod.rs [hunk 11]
```diff
@@ -1,4 +1,4 @@
-use failure::Error;
+use anyhow::Result;
```
Comment (GuillaumeGomez) [inline tests/buildtest/mod.rs]: Here³.

File: tests/buildtest/runner.rs [hunk 12]
```diff
@@ -1,9 +1,9 @@
-use failure::Error;
+use anyhow::Result;
```
Comment (GuillaumeGomez) [inline tests/buildtest/runner.rs]: Here⁴++.

File: tests/integration/crates_alt.rs [hunk 13]
```diff
@@ -1,10 +1,10 @@
-use failure::Error;
+use anyhow::Result;
```
Comment (GuillaumeGomez) [inline tests/integration/crates_alt.rs]: Here ⁷⁸⁵¹⁶⁴²²³¹⁰

File: tests/integration/crates_git.rs [hunk 14]
```diff
@@ -1,9 +1,9 @@
-use failure::Error;
+use anyhow::{anyhow, bail, Result};
```
Comment (GuillaumeGomez) [inline tests/integration/crates_git.rs]: I think you got it. XD

**Review by GuillaumeGomez — APPROVED** [review]

**Pull request merged**